### ISSUE TITLE
Setting some nods crashes the game (MP4)

### ISSUE THREAD

**Issue by CodyNinja1** [openplanet-nl/openplanet-issues#525]
Reproduction steps and example plugin can be found in the Openplanet discord in [this message sent in \#plugin-dev](https://discord.com/channels/276076890714800129/424967293538402334/1262337610735423541)

**Comment by CodyNinja1** [comment]
Per miss' request, here are the reproduction steps:

0. This is a Maniaplanet plugin, so open Maniaplanet
1. Go to Valley
2. If window isn't visible, enable the Openplanet interface by pressing F3 or w/e key you use to do that
3. Put `Wagon06` (as an example, Wagon01-08 all work) at the Wagon Name text input, and click `Get Wagon`
4. Go to `AccelCurve Properties`, and type `Loco` as the Donor Wagon Name (do note that `Loco` is the only Wagon with a non-null `AccelCurve`)
5. Click `Link AccelCurve from another Wagon using pure nods`, this will do the following:
```angelscript
auto DonorWagon = GetWagonFromFileName(DonorWagonFileName + ".Wagon.Gbx");
auto BorrowingWagon = wagon;
@BorrowingWagon.AccelCurve = @DonorWagon.AccelCurve;
```
After the final step, the game should crash.
Here is the plugin download: [Wagon-PlusPlus.zip](https://github.com/user-attachments/files/16233798/Wagon-PlusPlus.zip)
Here is a video showcasing the steps after step 1: https://streamable.com/ta8ei9
(it's a ZIP so I can upload it directly to github)

**Comment by codecat** [comment]
Thanks, this is fixed for the next update 🎉

**Comment by codecat** [comment]
The fix applied for this was broken - see #542. I am reverting this change and looking for a better solution.
